conjure-runtime/src/raw/mod.rs

```diff
@@ -66,9 +64,8 @@ where
 {
     type Response = T::Response;
     type Error = T::Error;
-    type Future = T::Future;
 
-    fn call(&self, req: R) -> Self::Future {
+    fn call(&self, req: R) -> impl Future<Output = Result<T::Response, T::Error>> {
```

Comment:
Why can't we do
```rust
async fn call(&self), req: R) -> Result<T::Response, T::Error>
```

Reply:
We could, but that would add an extra layer of indirection - it would end up making a wrapper future that looked something like

```rust
enum WrapperFuture<F> {
    Init,
    Awaiting(F),
}
```